extensions/ql-vscode/package.json

```diff
@@ -446,7 +446,7 @@
             "type": "boolean",
             "default": false,
             "scope": "application",
-            "markdownDescription": "Specifies whether to send CodeQL usage telemetry. This setting AND the global `#telemetry.enableTelemetry#` setting must be checked for telemetry to be sent to GitHub. For more information, see the [telemetry documentation](https://codeql.github.com/docs/codeql-for-visual-studio-code/about-telemetry-in-codeql-for-visual-studio-code)"
+            "markdownDescription": "Specifies whether to send CodeQL usage telemetry. This setting AND the one of the global telemetry settings (`#telemetry.enableTelemetry#` or `#telemetry.telemetryLevel#`) must be enabled for telemetry to be sent to GitHub. For more information, see the [telemetry documentation](https://codeql.github.com/docs/codeql-for-visual-studio-code/about-telemetry-in-codeql-for-visual-studio-code)"
```

Comment:
In the docs it says
> Tag your custom telemetry setting with `telemetry` and `usesOnlineServices` if you have one.

But I couldn't quite work out what this means. I need to hunt down the general docs for these settings to see how to add tags.

Reply:
I think this means that we need to add a `tags` property to our telemetry configuration options in the `package.json`. In [this documentation](https://code.visualstudio.com/api/references/contribution-points#contributes.configuration) for the configuration contribution points, they give this example:

```json
{
  "contributes": {
    "configuration": {
      "title": "Git",
      "properties": {
        "git.alwaysSignOff": {
          "type": "boolean",
          "scope": "resource",
          "default": false,
          "description": "%config.alwaysSignOff%"
        },
        "git.ignoredRepositories": {
          "type": "array",
          "default": [],
          "scope": "window",
          "description": "%config.ignoredRepositories%"
        },
        "git.autofetch": {
          "type": ["boolean", "string"],
          "enum": [true, false, "all"],
          "scope": "resource",
          "markdownDescription": "%config.autofetch%",
          "default": false,
          "tags": ["usesOnlineServices"]
        }
      }
    }
  }
}
```

The `tags` property doesn't seem to be documented on that page though, so I'm not sure what other tags there are.

Reply:
> I don't know what that means either. Do you have a link for where that's said?

I read it at https://code.visualstudio.com/api/extension-guides/telemetry#dos-and-donts

> In [this documentation](https://code.visualstudio.com/api/references/contribution-points#contributes.configuration) for the configuration contribution points, they give this example:

Thanks for finding that example. It's a shame there isn't more documentation on this but I think there's enough we can add the `tags` field. I believe it shouldn't cause any harm so we may as well add it.